### PR TITLE
improved retry

### PR DESCRIPTION
**Pull request by bbihari**
@doktoric @lalyos plz check

### CODE REVIEW

File: cli_tools [hunk 1]
```diff
@@ -10,6 +10,25 @@
...
 
+_retry() {
+  sleepsec=$1;
+  maxret=$2;
+  shift 2;
+  ret=1;
+  while :; do
+    [[ $ret > $maxret ]] && return 1;
+    echo "${ret}.retry...";
+    eval "$@" && return;
```
Comment (lalyos) [inline cli_tools]: adding a `|| exitCode=$?` at the end of eval, and followed by an `echo "---> command failed with exit code: $exitCode" 1>&2` would help to debug the issue what causes the retry

File: cli_tools [hunk 2]
```diff
@@ -10,6 +10,25 @@
...
 : ${AZURE_CLI_LOCATION:="/azure"}
 
+_retry() {
+  sleepsec=$1;
+  maxret=$2;
+  shift 2;
+  ret=1;
+  while :; do
+    [[ $ret > $maxret ]] && return 1;
+    echo "${ret}.retry...";
```
Comment (lalyos) [inline cli_tools]: please write to stderr. i know this strictscript doesn't really uses stderr. but this way the function is pretty much

File: cli_tools (hunk 1 above)
Comment (bbihari) [inline cli_tools]: Good idea. Fixed.

**Comment by lalyos** [comment]
LGTM

**Pull request merged**